compiler/src/iree/compiler/Dialect/LinalgExt/IR/TilingInterfaceImpl.cpp

```diff
@@ -2163,4 +2167,327 @@ LogicalResult OnlineAttentionOp::getResultTilePosition(
+  }));
+  return ranges;
+}
+
+SmallVector<Range> CustomOp::getIterationDomain(OpBuilder &builder) {
+  auto dims = llvm::to_vector(llvm::seq<unsigned>(0, getNumLoops()));
+  return getIterationDomainForDimensions(builder, dims,
+                                         /*symbols=*/ArrayRef<unsigned>{});
+}
+
+/// During tiling, the tiling implementation generates offsets and sizes
+/// to use for the dimensions of the operation that correspond to loops.
+/// The dimensions of the operation corresponding to symbols isnt tiled.
+/// This method extends the offsets and sizes for a tile with the
+/// offsets (which are zero) and sizes (which are same as the untiled sizes) for
+/// the dimension represented by symbols. The
+static std::pair<SmallVector<OpFoldResult>, SmallVector<OpFoldResult>>
+appendOffsetAndSizeForSymbolDimensions(OpBuilder &builder, CustomOp customOp,
+                                       ArrayRef<OpFoldResult> offsets,
+                                       ArrayRef<OpFoldResult> sizes) {
+  unsigned numSymbols = customOp.getNumNonLoopDimensions();
+  if (numSymbols == 0) {
+    return {llvm::to_vector(offsets), llvm::to_vector(sizes)};
+  }
+
+  auto appendedOffsets = llvm::to_vector(offsets);
+  appendedOffsets.resize(appendedOffsets.size() + numSymbols,
+                         builder.getIndexAttr(0));
+
+  auto appendedSizes = llvm::to_vector(sizes);
+  auto symbols = llvm::to_vector(llvm::seq<unsigned>(0, numSymbols));
+  auto symbolRanges = customOp.getIterationDomainForDimensions(
+      builder, /*dims=*/ArrayRef<unsigned>{}, symbols);
+  auto symbolRangesUb =
+      llvm::map_range(symbolRanges, [](Range r) { return r.size; });
+  appendedSizes.append(symbolRangesUb.begin(), symbolRangesUb.end());
+  return {appendedOffsets, appendedSizes};
+}
+
+/// This method is adapted from the method `linalg::computeAllSliceParameters`,
+/// adapted to work with map that have symbols, and empty maps.
+SmallVector<std::optional<linalg::SliceParameters>>
+computeCustomOpAllSliceParameters(OpBuilder &builder, Location loc,
+                                  CustomOp customOp, ValueRange valuesToTile,
+                                  ArrayRef<OpFoldResult> ivs,
+                                  SmallVector<OpFoldResult> tileSizes) {
+  assert(ivs.size() == static_cast<size_t>(llvm::count_if(
+                           llvm::make_range(tileSizes.begin(), tileSizes.end()),
+                           [](OpFoldResult v) { return !isZeroIndex(v); })) &&
+         "expected as many ivs as non-zero sizes");
+  unsigned numDims = customOp.getNumLoops();
+  unsigned numSymbols = customOp.getNumNonLoopDimensions();
+
+  // Construct (potentially temporary) mins and maxes on which to apply maps
+  // that define tile subshapes.
+  SmallVector<OpFoldResult> lbs =
+      linalg::computeTileOffsets(builder, loc, ivs, tileSizes);
+  SmallVector<OpFoldResult> sizeBounds = llvm::to_vector(tileSizes);
```

Comment:
nit: we don't need the `to_vector`.

```suggestion
  SmallVector<OpFoldResult> sizeBounds = tileSizes;
```